app/Models/Beatmapset.php

```diff
@@ -544,6 +566,26 @@ public function nominate(User $user)
         ];
     }
 
+    public function love(User $user)
+    {
+        if (!$this->isLoveable()) {
+            return [
+                'result' => false,
+                'message' => trans('beatmaps.nominations.incorrect_state'),
+            ];
+        }
+
+        $this->getConnection()->transaction(function () use ($user) {
+            $this->events()->create(['type' => BeatmapsetEvent::LOVE, 'user_id' => $user->user_id]);
+            $this->update(['approved' => static::STATES['loved']]);
+            $this->beatmaps()->update(['approved' => static::STATES['loved']]);
```

Comment:
this and the line above it should be replaced with `$this->setApproved('loved', $user)` (it also sets the approver id)